cartridges/int_algolia_controllers/cartridge/static/default/js/algolia/insights-config.js

```diff
@@ -25,33 +28,35 @@ function enableInsights(appId, searchApiKey) {
         }
     });
 
+    document.addEventListener('click', function (event) {
```

Comment:
Hello @sbellone ,
thank you for the fix. We may also refactor this code a bit like that:

```
function handleCartAction(eventName, queryID, objectID, indexName) {
    if (queryID && objectID && indexName) {
        window.aa('addedToCartObjectIDsAfterSearch', {
            eventName: eventName,
            index: indexName,
            queryID: queryID,
            objectIDs: [objectID],
        });
    }
}

// Global event listener for handling cart actions
document.addEventListener('click', function (event) {
    var queryID = getUrlParameter('queryID') || lastQueryID;
    var objectID = getUrlParameter('objectID') || lastObjectID;
    var indexName = getUrlParameter('indexName') || lastIndexName;

    if ($(event.target).is('button.add-to-cart')) {
        handleCartAction('Product Add to cart', queryID, objectID, indexName);
    }

    if ($(event.target).is('button#add-all-to-cart')) {
        handleCartAction('Global Add to cart', queryID, objectID, indexName);
    }
});```

So, we prevent registering 2 different event listeners and use some common logic.

Reply:
Oh yeah definitely thanks for the suggestion!